Modules/Core/Common/include/itkProgressReporter.h

```diff
@@ -62,14 +62,36 @@ class ITKCommon_EXPORT ProgressReporter
 public:
   ITK_DISALLOW_COPY_AND_ASSIGN(ProgressReporter);
 
-  /** Constructor sets progress to 0 because the filter is starting.  */
+  /** Old-style progress reporting. */
   ProgressReporter(ProcessObject * filter,
                    ThreadIdType    threadId,
                    SizeValueType   numberOfPixels,
-                   SizeValueType   numberOfUpdates = 100,
-                   float           initialProgress = 0.0f,
+                   SizeValueType   numberOfUpdates,
+                   float           initialProgress,
                    float           progressWeight = 1.0f);
 
+
+  /** New-style progress reporting. */
+  ProgressReporter(ProcessObject * filter,
+                   ThreadIdType,
+                   SizeValueType numberOfPixels,
```

Comment:
This in on location it will be very confusing combining the different behaviors the new-style expects the total number of "pixels" while the old expect the number just for the current threads.

Reply:
I now think it is best if we keep the new progress reporting separate. Trying to make it backwards compatible and elegantly plugging it into existing infrastructure seems like too much work.

Reply:
I am thinking about calling the new class `ProgressReporter2`, and defining it in the same header/source file as the original `ProgressReporter`. What do you think @blowekamp?